### ISSUE TITLE
[1.12.2] Release OptiFine 1.12.2 F4 (crtical bugfix)

### ISSUE THREAD

**Issue by Speiger** [sp614x/optifine#2975]
https://www.curseforge.com/minecraft/mc-mods/chunkpregenerator
![opera_2019-10-16_15-59-12](https://user-images.githubusercontent.com/2556269/66926240-f53b2780-f02d-11e9-9662-716a0e01e30d.png)

Could you finally after 10 months at least release critical fixes of a mod that is used in combo with Optifine Pretty much a lot. I get the same bugreport for at least 2-3 times a week just because you can not release a important "crashfix" within a reasonable time.

Referencing #2116

**Comment by Speiger** [comment]
Daily notify question.
When do you think its going to be released?

**Comment by realjackasterisk** [comment]
When it's done. Be patient.

**Comment by Speiger** [comment]
@sp614x i was patient 10 months.

**Comment by sp614x** [comment]
It will be out when it's ready.

**Comment by Speiger** [comment]
@sp614x yeah its been not 24 hours but for me its a new day. So daily reminder to finally release the mod update. Already making a plan to make it more annoying for you until you release it.
You had enough time. (Worst case I am just sending all the Chunkpregenerator people to you who want to report the crash to me, right now i am blocking them with this starting post info)

**Comment by sp614x** [comment]
I should probably delay the release in order to make a point that this is not the correct way to speedup development. Lets say 2 days delay per daily reminder post.

**Comment by Speiger** [comment]
Yeah just add as many days as you want. Patch for 1.12.2 that tells people to report issue reports to optifine is almost ready.

**Comment by Speiger** [comment]
@sp614x to add more to this:
You have in my view no right anymore to complain about rushing you.
I have dealt for 10 months with bugreports that you have caused for you without saying a thing.
So if I am a bit annoyed you may be understanding. So please add 10 months of delay. I happly do the same to you.

**Comment by sp614x** [comment]
The issue already has a TODO tag. It is not going to happen faster by creating drama.

**Comment by Speiger** [comment]
Ok. What stops you from doing it now. Is there something unstable? Or is there something that "HAS" to be in there before the release happens to ensure that other issues are not fixed. And can these be done within a week? If not then release it. It does not stop you from making another experimental patch.

**Comment by sp614x** [comment]
The current 1.12.2 F4 preview is waiting for 1.14.4 F4 to be released so they can have the same features.
The current 1.14.4 F4_pre11 seems stable enough and will probably be released in the next days. The 1.12.2 F4 will follow it shortly.

https://optifine.net/changelog?f=preview_OptiFine_1.12.2_HD_U_F4_pre3.jar
https://optifine.net/changelog?f=preview_OptiFine_1.14.4_HD_U_F4_pre11.jar

**Comment by Speiger** [comment]
Yeah then release F3 first and then update to F4 later when 1.14 is stable because forge is still developing. So for me this sounds like a endless development cycle.

**Comment by Speiger** [comment]
@sp614x patch is out. Have fun with the reports comming in.

**Comment by Velocihater** [comment]
I don't agree with hassling for development time, but I can see where Speiger is coming from. The only way his mod would be compatible is if people use the preview patch, which most don't know about. Not sure if I'm too keen on the whole calling for brigading though...

**Comment by realjackasterisk** [comment]
This kind of behavior is *completely* unacceptable. What a god damn child.

**Comment by Studio528-real** [comment]
Is there a way to mute or ban people from a github like there is discord? If not there should be. Shit comes when it comes, deal with it and stop being a whiney ass bitch and live life. Dont be a privileged brat

**Comment by Speiger** [comment]
I mean banning your userbase for asking for a "offical" release because of a fuckup that is your mistake does not seem really good.

**Comment by Studio528-real** [comment]
what I mean is that you're whining about, now just wait until the 1.12.2 pre comes out, once it does have fun, otherwise you probably should suffer tbh, you're not acting nice towards people and demanding something that is on a todo tag meaning its coming soon, just because f4_pre11 came out for 1.14.4 doesn't mean auto 1.12.2, he still has to backport it, which means he has to then code all the features into 1.12.2's framework instead of 1.14.4's

**Comment by Speiger** [comment]
Studio. You know how many versions he has skipped?
E4 should have been the fix release. But no he skipped until F4 to release his version.
He knew that there was a crash fix in there. But no he skipped knowingly 10months of work and at worst 5 versions, and when asked tell that it "Comes out soon" which means nothing.

And when asked more then once already showing how annoyed you are. They dare to increase intentionally the release date.

This bullcrap could have been solved by him very easily: "OK i am in the middle of the port but in 7 days you have your release". Before saying what he did or did not say until asked
Then nothing of this would happen.
So telling me: Dont whine about this bitch. Is not the best thing you could do.

Also just FYI: Did you deal for months with at least 1once a week for a "Your mod is crashing because of optifine" bugreports and helped people solve their problems for way to many months?
No you didnt. So shut the fuck up.

**Comment by Studio528-real** [comment]
Well simple a) dont use the mod and b) 1.14.4 is the main focus, so was 1.13.2. The latest non snapshot minecraft update always gets more priority over older version of the game such as 1.12.2, so could have even dropped 1.12.2 if he wanted, but no he kept updating it. But of people are going to whine, why should he give I.to whinney 12 y/o bastards who think that they can get whatever they want from someone over the internet. I've had games (such as minecraft) and had to wait a year or more for some bugs to be fixed. 10 months isnt really much tbh

**Comment by Speiger** [comment]
A) I dont use it but many others do. 
B) He already made the patch just didnt release it. 1 Tiny Change that takes 5 minutes. And no he didnt update Optifine because many many people dont know about preview releases. So they dont exist.
C) Well if he wants to be trusted as a developer then he should make proper work with fixes.
D) I worked a Fucking year to get my mod to the quality it has with massive work that is shared between 1.4.7-1.12.2 so i know the pain when requesting such things. But 1.12 is still one of the most used versions in Modded. So either he drops modded support completely or he just does the work he asked himself to do.
E) Quit bitching about people having at least a minimum of a standart.

**Comment by Studio528-real** [comment]
He cant just release an update for 1 thing... The pre's are updates and people may not know about them, but they can be told about them. Also who tf cares about a minimum of standard, he can take as long as he likes to. He is 1 man after all and 1.14.4 is the main version right now, so if he wants to focus on that instead of stuff from 10 months ago he can as he has the rights to. Let the man just fucking code and dont complain about shit over the internet, also why do you care if the mod is crashing if you don't even use it...

**Comment by Speiger** [comment]
Do you even understand what the issue we talk about?

**Comment by Studio528-real** [comment]
A seed preveiwer ingame that crashes because optifine "changes the integrated server" when it doesn't and only uses forge hooks, just use fucking admist if you want to see how a world looks.

**Comment by Speiger** [comment]
He expects a nullable to be always nonnull and thats why the game crashes. Thats a Vanilla thing. Nothing to do with forge or my mod. Just his fault

**Comment by Studio528-real** [comment]
If its vanilla then its not an optifine issue...

**Comment by Speiger** [comment]
No Then vanilla would crash but it does not. Optifine just changes the "Fixed" to the "Broken" state.

**Comment by Speiger** [comment]
Any other mod that creates a custom server instance and has its own starter would crash with optifine because optifine expects something to be "Always Present" which vanilla Declared as "Can be missing".

**Comment by Studio528-real** [comment]
Tbh why does this mod matter. It likely just puts more stress on gpu and cpu to do this while admist (it might be an external program) does not put stress on cpu or GPU. I expect this mod to at least put a lot more stress on cpu.

**Comment by Speiger** [comment]
Oh maybe because you actually have a accurate view of how the world would look like?
Admist is only like 80%-90% Vanilla accurate.
Why would you use optifine +1.12? Its made clear by youtubers that Optifine has a performance drop instead a increase.

**Comment by Studio528-real** [comment]
It may drop fps a little but mind you its still a hell of a lot better than playing eoth 5 fps for people, PLUS a lot of resource packs have support for it such as a few of my own personal packs, dokucraft, a few others out there, plus you get shaders and all that

**Comment by Speiger** [comment]
https://youtu.be/fECmwOUj2XA?t=159

**Comment by Studio528-real** [comment]
Plus youtubers are used to like 600+ fps so if there is any drop they tend to freak the fuck out

**Comment by Speiger** [comment]
Anyways. You didnt get the topic. Optifine should finally release the fix offically because the Previews are hidden enough that they are not seen by the players that have the issues.
So stop trolling around.

**Comment by Studio528-real** [comment]
Plus a base like that will always be a drop to fps. He probably has some settings that make it drop more too anyways.

**Comment by Purps4** [comment]
...
If you believe a fix is that important that requires you to bad mouth the only dev of optifine. May I suggest that you either: link the preview versions from the optifine site on your mod page or help users that are reporting bugs for your mod to consider downloading the preview version and giving them directions on how to do it. If you're so aware that other people don't know about the preview versions, why not tell them about it? An official release will get more attention, yes, but it is a fault of your own if you don't know how to or actually properly inform users of your mod to download the preview versions, as they fix glaring issues that the latest release of optifine 1.12.2 appears to have.

I'm a little disappointed that a mod developer would proceed to be this immature about a genuine issue that's on the way to being fixed. It's much more constructive to consider being more informative than to push issues onto someone potentially unrelated.

**Comment by Speiger** [comment]
@Purps4 look at the pregenerator comment section. Every person who reported was shown how to fix their issue. The self help section was with info to help people.

After a long time doing that to every single person manually I am sick of it.
Implementing in mod a auto link version wouldnt work because if he removes that version or a newer version comes out i would be destributing wrong versions.

About bad mouthing. Only started when they gave me a reason to do so. 
Aka giving me the reason to think nothing will happen. And the doubling down on it.
he only went back when i started showing that i was willing to go far with this.
In other words when it was to late. I did my part long enough. Now its his time to shine

**Comment by Studio528-real** [comment]
dude, just provide a link to a version and if a new is released then update it, its not that hard

**Comment by Speiger** [comment]
Dude. Or ask @sp614x to release the right version.

**Comment by Speiger** [comment]
Also this "Auto Link to report issues" shuts itself down automatically as soon the version was released properly

**Comment by Speiger** [comment]
https://cdn.discordapp.com/attachments/403255431373389846/634932315801387010/opt.png

**Comment by Studio528-real** [comment]
he doesn't have to, he can when he wants, just tell people to download the pre-releases instead of bitching and whining about an "incorrect update and having to wait 10 months". now stop being immature about it and just wait another 10 months or tell people to use pre versions (no it will probably release sooner but just calm your fucking tits). its like a game dev, asking them to release a version isnt going to do it any faster, so why would that not apply to a mod dev too...

**Comment by Studio528-real** [comment]
requesting people to also rush a dev will not get it done faster also, best to wait it out as you have done and put a friendly message of "Hey, there are some bugs in this version of optifine, please update to a prerelease version of it for these not to happen. or you can wait until the dev updates to a full release."

**Comment by Speiger** [comment]
I am not asking him to rush to create something new. I am asking him to move something in the spotlight because even so I did what you suggested (telling in the known issues section how to fix the optifine crash) I got still reports. So since I get constant requests how to fix this I am now moving that to here. And I will continue until its properly out.
And since you have no idea what you talk about why am I feeding a troll who wants to improve optifines users live

**Comment by Studio528-real** [comment]
hahahahahaha. the way you have worded the whole thing and "daily reminder" and all of that has indicated that you are trying to bring something to his attention and tell him to release it right now so you stop getting bugged, just ignore those people since they are too dumb to read anyways... if they cant see that you said how to fix it, don't bother with their time or just tell them to read that or even link issues to them of the other people asking... im just laughing at how you are acting, and im trying to tell you that taking a whinney bratty attitude towards a dev will not get you anywhere in life, whether IRL or online, your best bet is to just let sp do his shit and work on coding and making sure he gets a stable of 1.14.4 out before he worries about older versions of minecraft.

**Comment by Studio528-real** [comment]
![image](https://user-images.githubusercontent.com/52116771/67137603-58c38180-f1f5-11e9-95a6-b50053e75e07.png)
also, he gave you pretty much an exact time frame on when it will come out, likely this week so stop bitching about it

**Comment by Studio528-real** [comment]
> 
> 
> The current 1.12.2 F4 preview is waiting for 1.14.4 F4 to be released so they can have the same features.
> The current 1.14.4 F4_pre11 seems stable enough and will probably be released in the next days. The 1.12.2 F4 will follow it shortly.
> 
> https://optifine.net/changelog?f=preview_OptiFine_1.12.2_HD_U_F4_pre3.jar
> https://optifine.net/changelog?f=preview_OptiFine_1.14.4_HD_U_F4_pre11.jar
> 
> The 1.14.4 F4 took too long to stabilize, mostly due to Forge compatibility problems, but at this time 1.12.2 was already on the F4 feature branch, so it couldn't be released as F3.

also, he gave you pretty much an exact time frame on when it will come out, likely this week so stop bitching about it

**Comment by Studio528-real** [comment]
also telling people to spam the issues is also not going to help btw, been scrolling through your stupidity.  
![image](https://user-images.githubusercontent.com/52116771/67137686-4b5ac700-f1f6-11e9-9c23-286e15938ca0.png)

**Comment by Purps4** [comment]
> https://cdn.discordapp.com/attachments/403255431373389846/634932315801387010/opt.png

This picture sums up your maturity and informal assistance you're willing to give to users of your mod. That's genuinely upsetting, uninformative, and does not help users enjoy your mod in the best way possible, as soon as possible. **You have to work with Optifine. Optifine will not work with you.** Until you understand that, I'm going to continue feeling sorry for the people that genuinely appreciate your mod, as your competence and respect as a mod developer has dropped immensely in the Optifine community just within this single thread.

**Comment by SpGilbert** [comment]
can everyone just wait for once

**Comment by Speiger** [comment]
Another day. Another reminder @sp614x.
Finally release the fixes of Optifine.
To make sure that you don't forget this for another 10 months or skip this for another 10 months.

**Comment by TimeHawk** [comment]
OMFG mate just leave it alone, this will not help anyone or anything.

**Comment by bageldotjpg** [comment]
Spieger, you look like a clown.  This is just pathetic. They will get to it

**Comment by Speiger** [comment]
@Timehawk then its no problem that I send everyone here because I got reports for people not finding the preview versions. Many people said to me they didn't know about it. So I am fine with continuing it.

Also no the fact that its in the preview versions makes it so bad. He fixes something then hides it for months.
Not fixing something for months can have reasons. Hiding fixes not.

**Comment by TimeHawk** [comment]
You should not harass Sp (the only dev) to "spotlight" a version because your users are so uneducated they cant find a button and about the whole "hiding" thing it's not "hiding" as anyone who actually knows anything about OF knows about the preview versions(you've also been given an eta, so stop whining). At this point your just being an entitled ass, the world does not revolve around you or your mod and your issue is minuscule. Like everyone has said (I'm aware you have done this) just tell your users to download the preview in **big bold letters** on your post or say it's not compatible with OF, this has nothing to do with Sp "hiding" a version, it's only the fact that you're annoyed at **your** users and too lazy to respond to them. When will this get into your head?

**Comment by Studio528-real** [comment]
look dude, there is an ETA of sometime after 10/20/19 and its on the TODO tag. he will get to it, but I feel like it's now going to be delayed by another 2 days since you gave another daily reminder post. stop being a bitch. if people cant look TELL THEM TO LOOK don't just be an asshole to the dev and people. just tell them in a nice way instead.

**Comment by Speiger** [comment]
Well its funny that @sp614x hasnt locked this thread or banned me. 
No he decided to say nothing and let other people speak for him. 
Those people decided to insult the Optifine Userbase without realising it.

> You should not harass Sp (the only dev) to "spotlight" a version because your users are so uneducated they cant find a button and about the whole "hiding" thing it's not "hiding" as anyone who actually knows anything about OF knows about the preview versions(you've also been given an eta, so stop whining)

But if you think about it. Locking this thread or banning me would basically say that he agress with me.
The only thing he can do and is doing right now is. Shut up and do as quickly as he can before this bullshit gets worse then those Other people fanboying around without actually reading what the issue is. Because they skipped almost every argument and throw out bullshit.

I think this should be enough fuel for you guys to blow up over.

**Comment by sp614x** [comment]
@Speiger the video that you have linked compares:
- OptiFine with Anisotropic Filtering (AF) and render distance 16 vs 
- Vanilla without AF and render distance 14

AF itself (due to how MC engine works and the tricks that OF has to use to make AF work) can seriously decrease FPS. Add the higher render distance on top and no wonder the FPS is lower.

**Comment by Speiger** [comment]
@sp614x finally back. How is the release comming?

**Comment by sp614x** [comment]
Added 2 days to the release day (after it is ready to be shipped).

**Comment by Speiger** [comment]
Please add another 300 days if you want to. It just gives more people time to react to this.

**Comment by bageldotjpg** [comment]
You realize if anyone sees this and reacts to this they will think you are acting like an entitled jerk, Spieger?

**Comment by MrCheeze446** [comment]
@Speiger patience: the capacity to accept or tolerate delay, trouble, or suffering without getting angry or upset.

**Comment by Speiger** [comment]
@MrCheeze446 would you agree that waiting 300 days is enough patients?

**Comment by MrCheeze446** [comment]
@Speiger please pay attention to the part that says "without getting angry or upset"

**Comment by RetroBot98** [comment]
Optifine is a large mod, it takes time to develop it and properly fix all (or at least most of) the bugs, you complaining, won't make it go any faster, you and everyone else has to wait and be patient untill its ready, just calm down and be patient

**Comment by bageldotjpg** [comment]
@Speiger *patience

**Comment by Speiger** [comment]
@RetroBot98 the fix (in prerelease form) is out since 300 days. Multiple potential offical releases have been skipped.

@MrCheeze446 what says that I am angry with not having it released? I just want a exact date, no "comming soon" because the oldest Todos are 2 years old. (Just check git)
And the only thing that annoys me is this bullshit about the arguments that ignore the actual issue.
And the self insults are just weird.

If @sp614x gives me a clear date. The Questions are gone. Hasn't done so far.

**Comment by MrCheeze446** [comment]
@Speiger you seem very upset, you're demanding for a release date and you need to understand like many people have tried to tell you OptifFne is a large mode and takes time. Sp is currently focusing his time into an update for the 1.14 version of the mod, as you can see here he just posted a preview version for it yesterday 
![image](https://user-images.githubusercontent.com/30207076/67148390-5cd6ba00-f25c-11e9-8c24-cdfd5c39452c.png)

**Comment by Speiger** [comment]
@MrCheeze446 and what did you miss about "Skipped Multiple Releases"?

**Comment by MrCheeze446** [comment]
@Speiger skipped releases or not you are still coming off as very entitled and rude, demanding for a release date won't help you get any further, and several people arguing with you about it on Github won't do any good either. Overall you just need to wait, it may take a while but Sp has a life, he doesn't focus on OptiFine 24/7 so things will take time

**Comment by Speiger** [comment]
After cleaning up behing @sp614x mistakes isnt that reasonable?

**Comment by Speiger** [comment]
I mean wouldnt you be upset if i constantly shit at your house and other people get mad at you for asking not do it anymore?

**Comment by Speiger** [comment]
Because this is what is happening. I am asking him to stop sending me his userbase because of his fuckup. And he says: I am continuing a bit longer the more you ask?

**Comment by Studio528-real** [comment]
he isn't sending us. we are sending ourselves 
![image](https://user-images.githubusercontent.com/52116771/67148508-775d6300-f25d-11e9-8b08-ce62385bb11a.png) there is a GitHub bot that shows every post and comment on Github.

**Comment by Speiger** [comment]
Are you really that stupid?

**Comment by sp614x** [comment]
This tracker is for tracking development issues, not user feelings.
If the drama doesn't stop, the issue will be locked, label TODO removed and label DRAMA added.

**Comment by Speiger** [comment]
Oh yeah please do that. Because thats just more fuel for you.

**Comment by Speiger** [comment]
Because this tiny mob that follows you around is harming you right now quite a bit

**Comment by sp614x** [comment]
Thread locked due to lack of constructive feedback and too much drama.

**Comment by sp614x** [comment]
Continued in #2994
